### PR TITLE
gnomeExtensions.system-monitor: fix this package and upgrade to v33

### PR DESCRIPTION
**Pull request by tiramiseb**
###### Motivation for this change

Having a working extension :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

See the mailing list for a discussion about the sessionPath parameter in gnome3 (https://groups.google.com/forum/#!topic/nix-devel/jJevoZx2ZFo)...

### CODE REVIEW

File: pkgs/desktops/gnome-3/extensions/system-monitor/default.nix [hunk 1]
```diff
@@ -1,34 +1,41 @@
-{ stdenv, fetchFromGitHub, glib }:
+# This package needs the following configuration in configuration.nix:
+# services.xserver.desktopManager.gnome3.sessionPath = [ pkgs.libgtop pkgs.glib_networking ];
+{ config, stdenv, fetchFromGitHub, glib, glib_networking, libgtop, pkgs }:
 
 stdenv.mkDerivation rec {
   name = "gnome-shell-system-monitor-${version}";
-  version = "8b31f070e9e59109d729661ced313d6a63e31787";
+  version = "v33";
```
Comment (bjornfor) [inline pkgs/desktops/gnome-3/extensions/system-monitor/default.nix]: For Nix to understand the version, it must start with a number. (Change this one to "33" and set ```rev = "v${version}";``` below.)

**Review by bjornfor — APPROVED** [review]
Looks good. (Just nitpick about Nix version.)

**Comment by jtojnar** [comment]
Maybe it would be possible to patch the extension files to use [`g_irepository_prepend_search_path`](https://developer.gnome.org/gi/stable/GIRepository.html#g-irepository-prepend-search-path). Something like adding `imports.gi.GIRepository.Repository.prepend_search_path('@GI_TYPELIB_PATH@')` to the top of the `extension.js` file?

**Comment by tiramiseb** [comment]
Thanks for the idea @jtojnar ! I will take a look.

**Comment by jtojnar** [comment]
Not sure how safe it is wrt conflicts (e.g one extension adds `libfoo-X.Y.A` and other adds `libfoo-X.Y.B`). [Apparently](https://developer.gnome.org/gi/stable/GIRepository.html#g-irepository-get-default), there is only one GI repository per process. Maybe try testing it.

**Comment by tiramiseb** [comment]
I was going to try this approach, but I realized I don't know how I can retrieve the packages paths to include them with `prepend_search_path`...

**Comment by jtojnar** [comment]
You could use  `patches`  with `substituteAll`:

https://github.com/NixOS/nixpkgs/blob/7c4886d6859f328acd1646bc4e9432dc64419aeb/pkgs/applications/misc/orca/default.nix#L24-L29

**Comment by tiramiseb** [comment]
Great, thanks @jtojnar !
I have done it, and tested it. It works perfectly.

**Pull request merged**